src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldLookup.class.php

```diff
@@ -564,17 +564,21 @@ public function toString()
             return '';
         }
         $tableName = $this->GetConnectedTableName();
-        $nameColumn = 'name';
+        $identifyingColumnName = 'name';
```

Comment:
Every table should have a name column, it just doesn't suffice to hardcode `name` here. This should work (didn't test it though):

```php
$tableConf = new TCMSTableConf(TTools::GetCMSTableId($tableName));
$nameColumn = $tableConf->GetNameColumn();
```

Then we wouldn't need the fallback on the non-human-readable ID.

Reply:
Using $this->oDefinition->_oTableConf->GetNameColumn() now.